src/Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs

```diff
@@ -135,34 +133,64 @@ public static class WebApiServiceCollectionExtensions
         /// <returns></returns>
         public static AuthenticationBuilder AddProtectedWebApi(
             this AuthenticationBuilder builder,
-            string configSectionName,
             IConfiguration configuration,
-            string jwtBearerScheme,
-            Action<JwtBearerOptions> configureOptions,
+            string configSectionName = "AzureAd",
+            string jwtBearerScheme = JwtBearerDefaults.AuthenticationScheme,
             X509Certificate2 tokenDecryptionCertificate = null,
             bool subscribeToJwtBearerMiddlewareDiagnosticsEvents = false)
         {
-            builder.Services.Configure(jwtBearerScheme, configureOptions);
-            builder.Services.Configure<MicrosoftIdentityOptions>(options => configuration.Bind(configSectionName, options));
+            return builder.AddProtectedWebApi(jwtBearerScheme,
+                                            options => configuration.Bind(configSectionName, options),
+                                            options => configuration.Bind(configSectionName, options),
+                                            tokenDecryptionCertificate,
+                                            subscribeToJwtBearerMiddlewareDiagnosticsEvents);
+        }
+
+        /// <summary>
+        /// Protects the Web API with Microsoft identity platform (formerly Azure AD v2.0)
+        /// This method expects the configuration file will have a section, named "AzureAd" as default, with the necessary settings to initialize authentication options.
+        /// </summary>
+        /// <param name="builder">AuthenticationBuilder to which to add this configuration</param>
+        /// <param name="configSectionName">The configuration section with the necessary settings to initialize authentication options</param>
+        /// <param name="configuration">The Configuration object</param>
+        /// <param name="jwtBearerScheme">The JwtBearer scheme name to be used. By default it uses "Bearer"</param>
+        /// <param name="jwtOptionsMapper">An action to configure JwtBearerOptions</param>
+        /// <param name="tokenDecryptionCertificate">Token decryption certificate</param>
+        /// <param name="subscribeToJwtBearerMiddlewareDiagnosticsEvents">
+        /// Set to true if you want to debug, or just understand the JwtBearer events.
+        /// </param>
+        /// <returns></returns>
+        public static AuthenticationBuilder AddProtectedWebApi(
+        this AuthenticationBuilder builder,
+        string jwtBearerScheme,
+        Action<JwtBearerOptions> jwtOptionsMapper,
+        Action<MicrosoftIdentityOptions> microsoftIdentityOptionsMapper,
+        X509Certificate2 tokenDecryptionCertificate = null,
+        bool subscribeToJwtBearerMiddlewareDiagnosticsEvents = false)
+        {
+            builder.Services.Configure(jwtBearerScheme, jwtOptionsMapper);
+            builder.Services.Configure<MicrosoftIdentityOptions>(microsoftIdentityOptionsMapper);
```

Comment:
I think the generic type can be removed, since it can be inferred or we can add one in the line above for consistency and if we want to be explicit or not. (Same applied to similar methods below).